src/runtime_src/core/pcie/emulation/common_em/system_utils.cxx

```diff
@@ -2,11 +2,18 @@
 
 namespace systemUtil {
 
-  inline void printErrorMessage(std::string command, int status)
-  {
-    if(!status)// no error
-      return;
-    std::cout << "ERROR: [EMU 60-600] " << command << " failed with the error code " << status << ". PLEASE CHECK YOUR PERMISSIONS " << std::endl;
+  inline void printErrorMessage(const std::string &command, int status)
+  {      
+    try { 
+      if (status) { 
+        throw status; 
+      } else {
+        return;
+      }
+    } 
+    catch (int status) { 
+      std::cout << "ERROR: [EMU 60-600] " << command.c_str() << " Exception Caught - Failed with the error code " << status << ". PLEASE CHECK YOUR PERMISSIONS " << std::endl;
+    }
```

Comment:
Hi @venkatp-xilinx Still makes no sense to me can you explain the reason for this try/catch?  It's a no-op, what's the point throwing and catching immediately?

Reply:
Thanks for the suggestion. My requirement is to immediate exit from the flow when any of the system operation fails. Now I have corrected the code accordingly